File: data/blocks/block_1/block_1_question_1/factorial.py
import sys

def factorial(number):
    result = 1
    for i in range(1, number + 1):
        result *= i
    return result

if __name__ == "__main__":
    if len(sys.argv) > 1:  # Verifica si se proporcionó un argumento
        try:
            num = int(sys.argv[1])  # Convierte el argumento a entero
            print(factorial(num))
        except ValueError:
            print("Error: Ingresa un número entero válido.")
    else:
        print("Error: Debes ingresar un número como argumento.")

File: users_input/a/block_1/block_1_question_1/factorial.c
#include <stdio.h>
#include <stdlib.h>

int factorial(int number) {
    int result = 1;
    for (int i = 1; i <= number; ++i) {
        result *= i;
    }
    return result;
}

void processFile(const char* filename) {
    FILE* file = fopen(filename, "r");
    if (file == NULL) {
        fprintf(stderr, "Error: No se pudo abrir el archivo.\n");
        return;
    }

    char line[256];
    while (fgets(line, sizeof(line), file)) {
        int number = atoi(line); // Convertir la línea a entero
        if (number < 0) {
            fprintf(stderr, "Error: el número debe ser positivo.\n");
        } else {
            printf("%d\n", factorial(number));
        }
    }

    fclose(file);
}

int main(int argc, char* argv[]) {
    if (argc != 2) { // Verifica que se pase un archivo como argumento
        fprintf(stderr, "Uso: %s <archivo>\n", argv[0]);
        return 1;
    }

    processFile(argv[1]);
    return 0;
}
